marconi-chain-index/src/Marconi/ChainIndex/Indexers/Utxo.hs

```diff
@@ -263,16 +336,21 @@ eventIsBefore _ _                                                          = Fal
 
 -- | mappend, combine and balance Utxos
 instance Semigroup (StorableEvent UtxoHandle) where
-  (UtxoEvent us is cp) <> (UtxoEvent us' is' cp') =
-    UtxoEvent utxos txins (max cp cp')
-    where
-      toTxIn :: Utxo -> C.TxIn
-      toTxIn u = C.TxIn (u ^.txId) (u ^. txIx)
-      txins = Set.union is is'
+  (UtxoEvent us is cp) <> (UtxoEvent us' is' cp') = let
+
+      txins = Map.union is is'
+
+      insertUnspent :: Set Utxo -> Utxo -> Set Utxo
```

Comment:
:100: much cleaner

Reply:
And more resilient to future changes of the data structure, if needed.